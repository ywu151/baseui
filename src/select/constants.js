/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

export const OPTIONS = {
  single: 'single',
  multiple: 'multiple',
};

export const TYPE = {
  select: 'select',
  search: 'search',
};

export const ICON = {
  clearAll: 'clear_all',
  select: 'select',
  selected: 'selected',
  loop: 'loop',
};

export const STATE_CHANGE_TYPE = {
  select: 'select',
  unselect: 'unselect',
  clearAll: 'clear_all',
  keyDown: 'key_down',
  textChange: 'text_change',
};
