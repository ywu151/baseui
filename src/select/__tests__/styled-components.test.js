/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledInput,
  StyledInputContainer,
  StyledSelectComponentIcon,
  StyledDropDown,
  StyledDropDownItem,
  StyledOption,
} from '../index';
import {ICON, TYPE} from '../constants';

describe('Select styled components', () => {
  describe('StyledInput', function() {
    test('Input', () => {
      const component = shallow(
        <StyledInput>
          <div />
        </StyledInput>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledInput has correct styles',
      );
    });
  });
  describe('StyledInputContainer', function() {
    test('InputContainer', () => {
      const component = shallow(
        <StyledInputContainer>
          <div />
        </StyledInputContainer>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledInputContainer has correct styles',
      );
    });
  });

  describe('StyledSelectComponentIcon', () => {
    test.each([[ICON.clearAll], [ICON.select], [ICON.loop], [ICON.selected]])(
      '',
      $type => {
        const props = {$type};
        const component = shallow(
          <StyledSelectComponentIcon {...props}>
            <div />
          </StyledSelectComponentIcon>,
        );
        expect(component.instance().getStyles()).toMatchSnapshot(
          'StyledSelectComponentIcon has correct styles when type is ' + $type,
        );
      },
    );
  });

  describe('StyledDropDown', () => {
    test.each([
      ['', ''],
      ['$rows', 3],
      ['$type', TYPE.select],
      ['$isOpen', true],
    ])('', (prop, value) => {
      const props = {};
      props[prop] = value;
      const component = shallow(
        <StyledDropDown {...props}>
          <div />
        </StyledDropDown>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledDropDown has correct styles when ' + prop + ' is ' + value,
      );
    });
  });

  describe('StyledDropDownItem', function() {
    test('DropDownItem', () => {
      const component = shallow(
        <StyledDropDownItem>
          <div />
        </StyledDropDownItem>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledDropDownItem has correct styles',
      );
    });
  });

  describe('StyledOption', () => {
    test.each([[''], ['disabled'], ['$selected']])('', prop => {
      const props = {};
      props[prop] = true;
      const component = shallow(
        <StyledOption {...props}>
          <div />
        </StyledOption>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledOption has correct styles when ' + prop,
      );
    });
  });
});
