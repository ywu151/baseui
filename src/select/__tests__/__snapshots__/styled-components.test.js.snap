// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select styled components StyledDropDown : StyledDropDown has correct styles when  is  1`] = `
Object {
  "borderRadius": "$theme.sizing.scale300",
  "boxShadow": "$theme.lighting.shadow600",
  "display": "none",
  "height": null,
  "left": "$theme.sizing.scale300",
  "listStyle": "none",
  "overflowY": null,
  "position": "absolute",
  "top": null,
  "width": "calc(100% - $theme.sizing.scale600)",
}
`;

exports[`Select styled components StyledDropDown : StyledDropDown has correct styles when $isOpen is true 1`] = `
Object {
  "borderRadius": "$theme.sizing.scale300",
  "boxShadow": "$theme.lighting.shadow600",
  "display": null,
  "height": null,
  "left": "$theme.sizing.scale300",
  "listStyle": "none",
  "overflowY": null,
  "position": "absolute",
  "top": null,
  "width": "calc(100% - $theme.sizing.scale600)",
}
`;

exports[`Select styled components StyledDropDown : StyledDropDown has correct styles when $rows is 3 1`] = `
Object {
  "borderRadius": "$theme.sizing.scale300",
  "boxShadow": "$theme.lighting.shadow600",
  "display": "none",
  "height": "NaNpx",
  "left": "$theme.sizing.scale300",
  "listStyle": "none",
  "overflowY": "scroll",
  "position": "absolute",
  "top": null,
  "width": "calc(100% - $theme.sizing.scale600)",
}
`;

exports[`Select styled components StyledDropDown : StyledDropDown has correct styles when $type is select 1`] = `
Object {
  "borderRadius": "$theme.sizing.scale300",
  "boxShadow": "$theme.lighting.shadow600",
  "display": "none",
  "height": null,
  "left": "$theme.sizing.scale300",
  "listStyle": "none",
  "overflowY": null,
  "position": "absolute",
  "top": "$theme.sizing.scale600",
  "width": "calc(100% - $theme.sizing.scale600)",
}
`;

exports[`Select styled components StyledDropDownItem DropDownItem: StyledDropDownItem has correct styles 1`] = `
Object {
  "lineHeight": "$theme.sizing.scale600",
}
`;

exports[`Select styled components StyledInput Input: StyledInput has correct styles 1`] = `
Object {
  "::placeholder": Object {
    "color": "$theme.colors.mono700",
  },
  "backgroundColor": "transparent",
  "borderStyle": "none",
  "borderWidth": "0",
  "boxSizing": "border-box",
  "caretColor": "$theme.colors.primary",
  "color": "$theme.colors.mono1000",
  "cursor": "pointer",
  "flexGrow": "1",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "maxWidth": "100%",
  "outline": "none",
  "paddingBottom": "$theme.sizing.scale300",
  "paddingLeft": "$theme.sizing.scale500",
  "paddingRight": "$theme.sizing.scale500",
  "paddingTop": "$theme.sizing.scale300",
  "width": "auto",
}
`;

exports[`Select styled components StyledInputContainer InputContainer: StyledInputContainer has correct styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.mono200",
  "borderColor": "$theme.colors.mono700",
  "borderRadius": undefined,
  "borderStyle": "solid",
  "borderWidth": "1px",
  "boxShadow": "0 2px 6px transparent",
  "boxSizing": "border-box",
  "color": "$theme.colors.mono1000",
  "display": "flex",
  "flexWrap": "wrap",
  "fontFamily": "$theme.typography.font300.fontFamily",
  "fontSize": "$theme.typography.font300.fontSize",
  "fontWeight": "$theme.typography.font300.fontWeight",
  "lineHeight": "$theme.typography.font300.lineHeight",
  "paddingBottom": "0",
  "paddingLeft": "$theme.sizing.scale300",
  "paddingRight": "$theme.sizing.scale300",
  "paddingTop": "0",
  "position": "relative",
  "transitionDuration": "$theme.animation.timing100",
  "transitionProperty": "border, boxShadow, backgroundColor",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "width": "100%",
}
`;

exports[`Select styled components StyledOption : StyledOption has correct styles when  1`] = `
Object {
  ":hover": Object {
    "cursor": "pointer",
  },
  "color": null,
  "paddingLeft": "18px",
  "paddingRight": "18px",
}
`;

exports[`Select styled components StyledOption : StyledOption has correct styles when $selected 1`] = `
Object {
  ":hover": Object {
    "cursor": "pointer",
  },
  "color": "$theme.colors.primary400",
  "paddingLeft": "0px",
  "paddingRight": "0px",
}
`;

exports[`Select styled components StyledOption : StyledOption has correct styles when disabled 1`] = `
Object {
  ":hover": Object {
    "cursor": "not-allowed",
  },
  "color": "$theme.colors.mono700",
  "paddingLeft": "18px",
  "paddingRight": "18px",
}
`;

exports[`Select styled components StyledSelectComponentIcon : StyledSelectComponentIcon has correct styles when type is clear_all 1`] = `
Object {
  "cursor": "pointer",
  "marginLeft": "auto",
  "position": "absolute",
  "right": "$theme.sizing.scale600",
}
`;

exports[`Select styled components StyledSelectComponentIcon : StyledSelectComponentIcon has correct styles when type is loop 1`] = `
Object {
  "paddingLeft": "$theme.sizing.scale300",
}
`;

exports[`Select styled components StyledSelectComponentIcon : StyledSelectComponentIcon has correct styles when type is select 1`] = `
Object {
  "marginRight": "$theme.sizing.scale500",
  "position": "absolute",
  "right": "$theme.sizing.scale600",
  "top": "50%",
}
`;

exports[`Select styled components StyledSelectComponentIcon : StyledSelectComponentIcon has correct styles when type is selected 1`] = `
Object {
  "paddingRight": "$theme.sizing.scale300",
}
`;
